tests/subsys/pm/power_residency_time/testcase.yaml

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+tests:
+  pm.power_residency_time:
+    platform_allow:
+      - stm32l562e_dk
+    tags:
+      - pm
+    harness: pytest
+    harness_config:
+      pytest_dut_scope: session
+      pytest_args: [--powershield=<path_to>/STMicroelectronics_PowerShield__Virtual_ComPort,
+                    --expected-values=<path_to>/stm32l562e_dk.yaml]
+  pm.power_residency_time.pm_device:
+    platform_allow:
+      - stm32l562e_dk
+    tags:
+      - pm
+    extra_args:
+      - CONFIG_PM_DEVICE=y
+    harness: pytest
+    harness_config:
+      pytest_dut_scope: session
+      pytest_args: [--powershield=<path_to>/STMicroelectronics_PowerShield__Virtual_ComPort,
+                    --expected-values=<path_to>/stm32l562e_dk.yaml]
```

Comment:
IIUC these need to be set manually before testing, right ?

Any way we can spare the need for manual update ?
At least, `<path_to>/stm32l562e_dk.yaml` is rather static, so it looks reachable.
For `<path_to>/STMicroelectronics_PowerShield__Virtual_ComPort` maybe a dedicated twister configuration could be used or added  ?